augur/tasks/github/facade_github/tasks.py

```diff
@@ -190,8 +190,9 @@ def link_commits_to_contributor(session,contributorQueue):
                 UPDATE commits 
                 SET cmt_ght_author_id=:cntrb_id
                 WHERE 
-                cmt_author_raw_email=:cntrb_email
-                OR cmt_author_email=:cntrb_email
+                (cmt_author_raw_email=:cntrb_email
+                OR cmt_author_email=:cntrb_email)
+                AND cmt_ght_author_id is NULL
```

Comment:
@IsaacMilarky : Without that `AND cmt_ght_author_id is NULL` we are updating a *lot* more rows than we need to *every* time. I think this helps explain why the commits table started to bloat. I checked this with the following SQL `select * from commits where cmt_ght_author_id is NULL limit 1000` and 
```sql
SELECT
	* 
FROM
	commits 
WHERE
  cmt_ght_author_id IS NULL AND
	( cmt_author_raw_email = 'dev@shardingsphere.apache.org' OR cmt_author_email = 'dev@shardingsphere.apache.org' ) 
	LIMIT 10
```

We will be skipping a significant number of costly updates.

Reply:
Put another way: Every time postgresql does an update on a row, it copies the row to new physical space that's slowly recovered with autovacuum. I strongly suspect that by updating every row, even if it was already populated, was creating a ton of dead rows on the commits table very quickly. My working theory is that this will fix this problem. I am going to test that theory around March 29.

Reply:
That makes sense to me. This looks much better.